tests/unit_tests/test_ustar_cp/test_cpdFmax2pCp3.py

```diff
@@ -0,0 +1,407 @@
+from oneflux_steps.ustar_cp_python.cpdFmax2pCp3 import cpdFmax2pCp3, calculate_p_high, calculate_p_low, interpolate_FmaxCritical
+
+testcases = [
+        # Fmax = NaN
+        (np.nan, 52, np.nan),
+        # n = NaN
+        (42, np.nan, np.nan),
+        # n < 10
+        (3.14159265358979323, 9, np.nan),
+        # Fmax Below f-critical(1)
+        (5, 20, 0.4393355161666135),
+        (5.45204127574611, 52, 0.384643400326067), 
+        # Fmax Above f-critical(3)
+        (30, 1000, 2.7278340575698223e-05),
+        (17, 53, 0.005441758015001),
+        # Between f-critical(1) and f-critical(3)
+        (10, 52, 0.0761404222166437),
+        # fmax = 0
+        (0, 55, 1.0),
+        # fmax = fcritical(1)
+        (1.6301, 52, 1.0),
+        # Nominal
+        (2.37324492970613, 55, 1.0),
+        (10.3567400792636, 54, 0.0657053181314848)
+]
+@pytest.mark.parametrize("fmax, n, expected_p3", testcases)
+def test_cpdFmax2pCp3_matlab(matlab_engine, fmax, n, expected_p3):
+    """
+    Test the cpdFmax2pCp3 function in MATLAB.
+    Args:
+        matlab_engine (fixture): Fixture that initializes and manages the MATLAB engine session.
+        fmax, n : Matlab function input arguments
+        expected_p3 (float): The expected value of p3.
+    """
+    # Convert the input values to MATLAB compatible data types
+    fmax = matlab.double(fmax)
+    n = matlab.double(n)
+    expected_p3 = float(expected_p3)
+
+    # Run the MATLAB function
+    output_p3 = matlab_engine.cpdFmax2pCp3(fmax, n, nargout=1)
+
+    assert type(output_p3) == type(expected_p3), f"Result should be a {type(expected_p3)}, got {type(output_p3)}"
+    assert np.isclose(output_p3, expected_p3, equal_nan=True), f"Expected p3 value of {expected_p3}, but got {output_p3}."
+
+@pytest.mark.parametrize("fmax, n, expected_p3", testcases)
+def test_cpdFmax2pCp3(fmax, n, expected_p3):
+
+    output_p3 = cpdFmax2pCp3(fmax, n)
+
+    assert type(output_p3) == type(expected_p3), f"Result should be a {type(expected_p3)}, got {type(output_p3)}"
+    assert np.isclose(output_p3, expected_p3, equal_nan=True), f"Expected p3 value of {expected_p3}, but got {output_p3}."
+
+
+@pytest.mark.parametrize(
+"Fmax, FmaxCritical_high, n, expected_p",
+[
+    (20, 15, 50, 0.0018068999227986993),
+    (15, 15, 50, 0.010000000000000009),
+    (15.1, 15, 50, 0.00965419741276663),
+    (30, 15, 100, 4.4960217949308046e-05),
+],
+)
+
+def test_calculate_p_high_return_numeric_matlab(matlab_engine, Fmax, FmaxCritical_high, n, expected_p):
+    """
+    Test the cpdFmax2pCp3 MATLAB function for cases where numeric values are returned.
+    """
+    # Handle NaN inputs for MATLAB
+    Fmax_matlab = matlab.double([Fmax]) if not np.isnan(Fmax) else matlab.double([float('nan')])
+    FmaxCritical_high_matlab = matlab.double([FmaxCritical_high]) if not np.isnan(FmaxCritical_high) else matlab.double([float('nan')])
+    n_matlab = matlab.double([n]) if not np.isnan(n) else matlab.double([float('nan')])
```

Comment:
Just ` n_matlab = matlab.double([n]) ` is sufficient. 
When `n=np.nan`: 
` matlab.double([n])` is equivalent to `matlab.double([float('nan')])`